PortalSwift/Sources/PortalSwift/Core/Api/PortalApi.swift

```diff
@@ -0,0 +1,589 @@
+//
+//  PortalApi.swift
+//
+//  Created by Portal Labs, Inc.
+//  Copyright © 2022 Portal Labs, Inc. All rights reserved.
+//
+
+import Foundation
+
+/// The class to interface with Portal's REST API.
+public class PortalApi {
+  private var _client: ClientResponse?
+  private var apiKey: String
+  private var baseUrl: String
+  private let decoder = JSONDecoder()
+  private var httpRequests: HttpRequester
+  private let logger = PortalLogger()
+  private var provider: PortalProvider?
+  private let requests: PortalRequests
+  private let featureFlags: FeatureFlags?
+
+  private var address: String? {
+    self.provider?.address
+  }
+
+  private var chainId: Int? {
+    self.provider?.chainId
+  }
+
+  public var client: ClientResponse? {
+    get async throws {
+      if self._client == nil {
+        self._client = try await self.getClient()
+      }
+      return self._client
+    }
+  }
+
+  /// Create an instance of a PortalApi class.
+  /// - Parameters:
+  ///   - apiKey: The Client API key. You can create one using Portal's REST API.
+  ///   - apiHost: (optional) The Portal API hostname.
+  ///   - provider: The PortalProvider instance to use for stateful Provider info (chainId, address)
+  public init(
+    apiKey: String,
+    apiHost: String = "api.portalhq.io",
+    provider: PortalProvider? = nil,
+    featureFlags: FeatureFlags? = nil,
+    requests: PortalRequests? = nil
+  ) {
+    self.apiKey = apiKey
+    self.baseUrl = apiHost.starts(with: "localhost") ? "http://\(apiHost)" : "https://\(apiHost)"
+    self.featureFlags = featureFlags
+    self.provider = provider
+    self.requests = requests ?? PortalRequests()
+    self.httpRequests = HttpRequester(baseUrl: self.baseUrl)
+  }
+
+  /*******************************************
+   * Public functions
+   *******************************************/
+
+  public func eject() async throws -> String {
+    if let url = URL(string: "\(baseUrl)/api/v3/clients/me/eject") {
+      do {
+        let data = try await post(url, withBearerToken: self.apiKey)
+        guard let ejectResponse = String(data: data, encoding: .utf8) else {
+          throw PortalApiError.unableToReadStringResponse
+        }
+
+        return ejectResponse
+      } catch {
+        self.logger.error("PortalApi.getBalances() - Unable to eject: \(error.localizedDescription)")
+        throw error
+      }
+    }
+
+    self.logger.error("PortalApi.eject() - Unable to build request URL.")
+    throw URLError(.badURL)
+  }
+
+  public func getBalances(_ chainId: String) async throws -> [FetchedBalance] {
```

Comment:
`chainId` here is CAIP-2 right?

Reply:
Yes

Reply:
Can we rename this method to `getErc20Balances`?